apple/Sources/FerrostarCore/NavigationState.swift

```diff
@@ -19,22 +19,50 @@ public struct NavigationState {
     public internal(set) var isCalculatingNewRoute: Bool = false
 
     init(snappedLocation: CLLocation, heading: CLHeading? = nil, fullRoute: [CLLocationCoordinate2D], steps: [RouteStep]) {
-        self.snappedLocation = snappedLocation
-        self.heading = heading
-        self.courseOverGround = snappedLocation.course
-        self.fullRouteShape = fullRoute
+        self.snappedLocation = UserLocation(clLocation: snappedLocation)
+        if let heading {
+            self.heading = Heading(clHeading: heading)
+        }
+        self.courseOverGround = CourseOverGround(course: snappedLocation.course,
+                                                 courseAccuracy: snappedLocation.courseAccuracy)
+        self.fullRouteShape = fullRoute.map { GeographicCoordinate(lat: $0.latitude, lng: $0.longitude) }
```

Comment:
I think you may have added a convenience init in the extensions that cleans this up slightly.